docs/DeveloperGuide.md

```diff
@@ -1479,10 +1491,112 @@ testers are expected to do more *exploratory* testing.
+   1. Test case: Invalid Prefix or missing compulsory Prefixes e.g. `edit ct/s i/1 n/Alice Pte Ltd p/90345623 e/alice@gmail.com z/large `
+      or `edit ct/s i/1`<br>
+      Expected: No supplier is added. Error details shown in the response message. A help message for edit command will also be displayed
+      to guide user accordingly. SupplierList on GUI remains unchanged.
+   1. Test case: Edits a supplier with existing SUPPLIER_NAME in list e.g. `edit ct/s i/1 n/Alice Pte Ltd` followed by `edit ct/s i/2 n/Alice Pte Ltd`<br>
+      Expected: An error will occur and a message will be displayed, stating that a supplier with duplicate
+      SUPPLIER_NAME cannot be added into the list. SupplierList on GUI remain unchanged.
+
+### Editing a Warehouse
+
+1. Edit command format: `edit ct/w i/INDEX [n/NAME] [p/PHONE] [addr/ADDRESS] [r/REMARK]`
+
+    1. Prerequisites: Warehouses in CLI-nic does not have a warehouse named Alice Warehouse (with the exception of test case to test for duplicated warehouse).
+
+   1. Test case: Minimal information e.g. `edit ct/w i/1 n/Alice Warehouse`<br>
+      Expected: Edits a warehouse in index 1 on warehouse list to have a name "Alice Warehouse".
+   1. Test case: With all fields supplied e.g. `edit ct/w i/1 n/Alice Warehouse p/82345162 addr/21 Lower Kent Ridge Rd, Singapore 119077 r/Largest Warehouse`<br>
+      Expected: Edits a warehouse in index 1 on warehouse list with all the fields applied.
+   1. Test case: Invalid Prefix or missing compulsory Prefixes e.g. `edit ct/w i/1 n/Alice Warehouse p/82345162 addr/21 Lower Kent Ridge Rd, Singapore 119077 z/large `
+      or `edit ct/w i/1`<br>
+      Expected: No warehouse is added. Error details shown in the response message. A help message for edit command will also be displayed
+      to guide user accordingly. WarehouseList on GUI remains unchanged.
+   1. Test case: Edits a warehouse with existing WAREHOUSE_NAME in list e.g. `edit ct/w i/1 n/Alice Warehouse` followed by `edit ct/w i/2 n/Alice Warehouse`<br>
+      Expected: An error will occur and a message will be displayed, stating that a warehouse with duplicate
+      WAREHOUSE_NAME cannot be added into the list. WarehouseList on GUI remain unchanged.
+
+## Exiting CLI-nic
+
+1. Exit command format: `exit`
+
+   1. Test case: Exit command with no additional arguments e.g. `edit`<br>
+      Expected: CLI-nic closes with current state of data saved.
+   1. Test case: Exit command with additional arguments e.g. `exit test` or `exit ct/s`<br>
+      Expected: Similar to previous.
+
+## Listing CLI-nic
+
+1. List command format: `list`
+
+   1. Test case: List command with no additional arguments e.g. `list`<br>
+      Expected: CLI-nic lists all suppliers and warehouses data in CLI-nic.
+   1. Test case: List command with additional arguments e.g. `list test` or `list i/1`<br>
+      Expected: Similar to previous.   
+
+### Viewing a Supplier
+
+1. View command format: `view ct/TYPE i/INDEX`
```

Comment:
Since we are splitting into viewing supplier and warehouse section, perhaps we could specify the type instead of leaving it as TYPE!

Reply:
Good idea! Shall we standardise for Add command as well :)

Reply:
I think it would be better!!